mycli/main.py

```diff
@@ -207,13 +217,16 @@ def read_my_cnf_files(self, files, keys):
                 pass
 
         sections = ['client']
+        if self.login_path and self.login_path != 'client':
+            sections.append(self.login_path)
+
         if self.defaults_suffix:
-            sections.append('client{0}'.format(self.defaults_suffix))
+            sections.extend([sect + self.defaults_suffix for sect in sections])
```

Comment:
I agree that it seems odd, but that is how `mysql` behaves (I just tested it again to double-check). [`--defaults-group-suffix' is not file-specific](https://dev.mysql.com/doc/refman/5.0/en/option-file-options.html#option_general_defaults-group-suffix), it adds that suffix to every option group that `mysql` looks at, even a login path group.

Login path option groups are just like regular option groups with two exceptions. 1) They are encrypted on the filesystem. 2) They only permit certain options: `host`, `user`, `password`, `port` and `socket`.